### PR TITLE
add delete group order button and confirmation modal

### PR DESCRIPTION
**Pull request by mwang2000**
image
image

### Description

-  Added a button to delete a group order on the View Order Modal and connected it to the thunk that deletes an order

- This was not in the task description but I decided to make a general confirmation modal that can be reused for any future confirmation popups (it only consists of a heading and a footer with 2 buttons with text that can be customized)

I thought this looked ok but feel free to suggest changes if it's bad UI/UX design!

### CODE REVIEW

**Review by MrBenC88 — APPROVED** [review]
looks good

File: web/src/components/ViewOrderModal.jsx [hunk 1]
```diff
@@ -34,6 +38,12 @@ export default chakra(function ViewOrderModal({
   isOpen,
   onClose,
 }) {
+  const { isOpen: isConfirmationOpen, onOpen: onConfirmationOpen, onClose: onConfirmationClose } = useDisclosure();
+  const dispatch = useDispatch();
+  const deleteOrder = () => {
+    dispatch(removeOrderAsync(data.orderId));
+    onConfirmationClose();
```
Comment (StevenZhao318) [inline web/src/components/ViewOrderModal.jsx:45]: This also should close the expanded modal for viewing order if delete is processed. Currently it opens modal for next group order

**Review by StevenZhao318 — CHANGES_REQUESTED** [review]
looks good, just a small fix needed

File: web/src/components/ViewOrderModal.jsx (hunk 1 above)
Comment (mwang2000) [reply web/src/components/ViewOrderModal.jsx:45]: @StevenZhao318 for some reason this doesn't happen to me, but I added an onClose() call to the ViewOrderModal on the next line, does that fix it?

**Review by StevenZhao318 — APPROVED** [review]
Nice work! Looks good

**Pull request merged**